mne/beamformer/tests/test_rap_music.py

```diff
@@ -0,0 +1,175 @@
+#
+# License: BSD (3-clause)
+
+import os.path as op
+import numpy as np
+
+import warnings
+from nose.tools import assert_true
+from numpy.testing import assert_array_almost_equal
+
+
+import mne
+from mne.datasets import testing
+from mne.beamformer import rap_music
+from mne.utils import slow_test
+
+
+data_path = testing.data_path(download=False)
+fname_raw = op.join(data_path, 'MEG', 'sample', 'sample_audvis_trunc_raw.fif')
+fname_cov = op.join(data_path, 'MEG', 'sample', 'sample_audvis_trunc-cov.fif')
+fname_fwd = op.join(data_path, 'MEG', 'sample',
+                    'sample_audvis_trunc-meg-eeg-oct-4-fwd.fif')
+fname_event = op.join(data_path, 'MEG', 'sample',
+                      'sample_audvis_trunc_raw-eve.fif')
+
+warnings.simplefilter('always')  # enable b/c these tests throw warnings
+
+
+def read_forward_solution_meg(fname_fwd, ch_names, **kwargs):
+    fwd = mne.read_forward_solution(fname_fwd, **kwargs)
+    return mne.pick_types_forward(fwd, eeg=False, include=ch_names,
+                                  exclude=['MEG 2443'])
+
+
+def _get_data(tmin=-0.1, tmax=0.15, event_id=1):
+    """Read in data used in tests
+    """
+    events = mne.read_events(fname_event)
+    raw = mne.io.Raw(fname_raw, preload=True)
+
+    # Setup for reading the raw data
+    raw.info['bads'] = ['MEG 2443']  # 2 bads channels
+
+    # Set up pick list: MEG - bad channels
+    picks = mne.pick_types(raw.info, meg=True, eeg=False, stim=False,
+                           eog=True, ref_meg=False, exclude='bads')
+
+    # Read epochs
+    epochs = mne.Epochs(raw, events, event_id, tmin, tmax, proj=True,
+                        picks=picks, baseline=(None, 0), preload=False,
+                        reject=dict(grad=4000e-13, mag=4e-12, eog=150e-6))
+
+    evoked = epochs.average()
+
+    ch_names = evoked.info['ch_names']
+    forward = mne.read_forward_solution(fname_fwd, ch_names)
+
+    forward_surf_ori = read_forward_solution_meg(fname_fwd, ch_names,
+                                                 surf_ori=True)
+    forward_fixed = read_forward_solution_meg(fname_fwd, ch_names,
+                                              force_fixed=True,
+                                              surf_ori=True)
+
+    noise_cov = mne.read_cov(fname_cov)
+
+    return evoked, noise_cov, forward, forward_surf_ori, forward_fixed
+
+
+def simu_data(evoked, forward, noise_cov, n_dipoles, times):
+    # Generate the two dipoles data
+    mu, sigma = 0.1, 0.005
+    s1 = 1 / (sigma * np.sqrt(2 * np.pi)) * np.exp(-(times - mu) ** 2 /
+                                                   (2 * sigma ** 2))
+
+    mu, sigma = 0.075, 0.008
+    s2 = 1 / (sigma * np.sqrt(2 * np.pi)) * np.exp(-(times - mu) ** 2 /
+                                                   (2 * sigma ** 2))
+    data = np.array([s1, s2]) * 10e-10
+
+    src = forward['src']
+    rndi = np.random.randint(len(src[0]['vertno']))
```

Comment:
you should be using a random state.

rng = np.random.RandomState(42)
rndi = rng.randint(len(src[0]['vertno']))

it's like having a local seed for random.